lib/logstash/filters/urldecode.rb

```diff
@@ -34,6 +37,13 @@ def filter(event)
     # If all_fields is true then try to decode them all
     if @all_fields
       event.to_hash.each { |name, value| event[name] = urldecode(value) }
+    # If a prefix is defined then try to decode all params starting with it
+    elsif !@prefix.nil?
+      event.to_hash.each do |name, value|
```

Comment:
I recommend  dry this a bit by using the select function (http://ruby-doc.org/core-2.2.2/Hash.html#method-i-select) of a hash. This would make the two options looks alike by doing in case of having the prefix, run the select and filter the hash and then just run the urldecode on the resulting fields.